src/test/java/upgrade/com/atlassian/bitbucket/jenkins/internal/UpgradeTest.java

```diff
@@ -115,7 +116,12 @@ private void compareFields(Set<Class<?>> releasedClasses, Map<String, Class<?>>
                 Field releasedField = entry.getValue();
                 Field unreleasedField = unreleasedFields.get(entry.getKey());
                 if (unreleasedField == null) {
-                    //field was removed, this should be ok
+                    String fullFieldName = releasedField.getDeclaringClass().getName() + '.' + releasedField.getName();
+                    if (safelyRemovedFields.containsKey(fullFieldName)) {
+                        // field has been marked as safely removed, so we can resolved
+                        i.remove();
```

Comment:
also need to remove it from `safelyRemovedFields` so that when `safelyRemovedFields` returns the list contains fields that were not actually referenced. We then fail the test if there are any in the list so that we keep this file up to date.